store/sqlstore/channel_store.go

```diff
@@ -2213,14 +2213,13 @@ func (s SqlChannelStore) GetFileCount(channelId string) (int64, error) {
 	var count int64
 	err := s.GetReplicaX().Get(&count, `
 		SELECT
-			COUNT(*)
+		    COUNT(FileInfo.*)
```

Comment:
It's slightly better to use `count(*)` rather than `count(fileinfo.*)`

Reply:
Note: TIL MySQL was struggling with that query, so I changed it as requested!

Reply:
Yes it does have an impact in performance. `count(*)` will just count the number of rows returned from the where condition, `count(id)` returns the number of rows with non-null `id`. That is the functional difference, but there is a difference in performance as well, in terms of index scans vs index-only scans.

For ex, if you are doing `select count(id) from fileinfo where deleteat=0`, it must do an index scan on the `deleteat` index. Whereas, if you do `select count(*) from fileinfo where deleteat=0`, it can get away with just an index-only scan.

If you are interested in more details, please do take a look at: https://docs.google.com/presentation/d/14LTiD4EHrlCgVfMFngnWe1Ov6gsbgwXMGp3b6jM5p3A/edit#slide=id.ge9b7e537ec_0_0
and the video at:
https://community.mattermost.com/core/pl/seqkudw3i7gi7jnn97xaj3nhuw

Reply:
Thanks for the resources Agniva!